core/src/components/datetime/utils/comparison.ts

```diff
@@ -38,13 +38,28 @@ export const isAfter = (baseParts: DatetimeParts, compareParts: DatetimeParts) =
   );
 };
 
-export const warnIfValueOutOfBounds = (value: DatetimeParts, min: DatetimeParts, max: DatetimeParts) => {
-  if ((min && isBefore(value, min)) || (max && isAfter(value, max))) {
+export const warnIfValueOutOfBounds = (
+  value: DatetimeParts | DatetimeParts[],
+  min: DatetimeParts,
+  max: DatetimeParts
+) => {
+  const isOutOfBounds = (value: DatetimeParts) => (min && isBefore(value, min)) || (max && isAfter(value, max));
+  const printWarning = () =>
     printIonWarning(
       'The value provided to ion-datetime is out of bounds.\n\n' +
         `Min: ${JSON.stringify(min)}\n` +
         `Max: ${JSON.stringify(max)}\n` +
         `Value: ${JSON.stringify(value)}`
     );
+
+  if (Array.isArray(value)) {
+    for (const val of value) {
+      if (isOutOfBounds(val)) {
+        printWarning();
+        break;
+      }
+    }
+  } else if (isOutOfBounds(value)) {
+    printWarning();
```

Comment:
Can we change this so that we do not have diverging code paths? Maybe we can do something similar to what `getCalendarDayState` does:

```js
const activePartsArray = Array.isArray(activeParts) ? activeParts : [activeParts];
```